include/itkWaveletCoeffsInverseFFTImageFilter.h

```diff
@@ -0,0 +1,132 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkWaveletCoeffsInverseFFTImageFilter_h
+#define itkWaveletCoeffsInverseFFTImageFilter_h
+
+#include "itkForwardFFTImageFilter.h"
+#include "itkInverseFFTImageFilter.h"
+#include "itkFFTPadImageFilter.h"
+#include "itkWaveletFrequencyForward.h"
+#include "itkWaveletFrequencyInverse.h"
+#include "itkWaveletFrequencyForwardUndecimated.h"
+#include "itkWaveletFrequencyInverseUndecimated.h"
+#include "itkWaveletFrequencyFilterBankGenerator.h"
+#include "itkHeldIsotropicWavelet.h"
+#include "itkVowIsotropicWavelet.h"
+#include "itkSimoncelliIsotropicWavelet.h"
+#include "itkShannonIsotropicWavelet.h"
+#include "itkMonogenicSignalFrequencyImageFilter.h"
+#include "itkVectorInverseFFTImageFilter.h"
+#include "itkPhaseAnalysisSoftThresholdImageFilter.h"
+#include "itkZeroDCImageFilter.h"
+#include "itkImage.h"
+#include "itkCastImageFilter.h"
+#include "itkNumberToString.h"
+#include <string>
+
+namespace itk
+{
+/** \class WaveletCoeffsInverseFFT
+ * @brief IsotropicWavelet multiscale coefficients extraction where input is an image in the spatial domain.
+ * Output Layout:
+ * Output m_TotalOutputs - 1 is the residual low pass filtered of the last level/scale.
+ * [N - 1]: Low pass residual, also called approximation.
+ * [0,..,HighPassBands): Wavelet coef of first level.
+ * [HighPassBands,..,l*HighPassBands]: Wavelet coef of l level.
+ *
+ * @note The information/metadata of input image is ignored.
```

Comment:
This can be avoided for this filter. You have the metadata of the spatial domain image (the input), and you are returning spatial domain images as well, we can restore the metadata. Set origin, spacing and direction of the outputs depending on the level.

In python:
```python
def restoreMetadata(imageToRestore, referenceImage, level):
    spacing = referenceImage.GetSpacing()
    origin = referenceImage.GetOrigin()
    direction = referenceImage.GetDirection()
    imageToRestore.SetSpacing(spacing * (2**level) )
    imageToRestore.SetOrigin(origin)
    imageToRestore.SetDirection(direction)
    return imageToRestore
```
And remember that the N-1 output (low pass residual) corresponds to the `l` level.

You can use this function to get the level (and the band) based on the output: https://github.com/InsightSoftwareConsortium/ITKIsotropicWavelets/blob/d8bdf5b29fa96a2cba2b6cf42d3395c3337d5623/src/itkWaveletUtilities.cxx#L26

Reply:
Ok, I see that you are using the undecimated transform, the size of the images is fixed in this case. So spacing does not change.

I see that you are using a ChangeInformationFilter to restore metadata. Remove the comment in the documentation about the metadata is lost.